taichi/codegen/codegen_llvm.cpp

```diff
@@ -1125,15 +1155,21 @@ void CodeGenLLVM::visit(GlobalLoadStmt *stmt) {
     llvm::Value *byte_ptr, *bit_offset;
     read_bit_pointer(llvm_val[stmt->ptr], byte_ptr, bit_offset);
     auto bit_level_container = builder->CreateLoad(builder->CreateBitCast(
-        byte_ptr, llvm::Type::getInt32PtrTy(*llvm_context)));
+        byte_ptr, llvm_ptr_type(cit->get_compute_type())));
     // 2. bit shifting
-    //    first left shift `32 - (offset + num_bits)`
-    //    then right shift `32 - num_bits`
+    //    first left shift `compute_type_size(like 32, 64, ...) - (offset +
+    //    num_bits)` then right shift `compute_type_size - num_bits`
+    auto compute_type_size = data_type_bits(cit->get_compute_type());
     auto bit_end = builder->CreateAdd(bit_offset,
                                       tlctx->get_constant(cit->get_num_bits()));
-    auto left = builder->CreateSub(tlctx->get_constant(32), bit_end);
-    auto right = builder->CreateAdd(tlctx->get_constant(32),
-                                    tlctx->get_constant(-cit->get_num_bits()));
+    auto left =
+        builder->CreateSub(tlctx->get_constant(compute_type_size), bit_end);
+    auto right = builder->CreateSub(tlctx->get_constant(compute_type_size),
+                                    tlctx->get_constant(cit->get_num_bits()));
+    left = builder->CreateIntCast(left, llvm_type(cit->get_compute_type()),
+                                  cit->get_is_signed());
+    right = builder->CreateIntCast(right, llvm_type(cit->get_compute_type()),
```

Comment:
After testing the case of 16-bits and 64-bits, I found that we should use the num of `physical_type` bits in `BitStructType` SNode here instead of the `compute_type` of `CustomIntType`. So I decided to convert this pr to draft and reimplement this part.